cpp/models/abm/testing_strategy.cpp

```diff
@@ -175,15 +178,15 @@ bool TestingStrategy::run_strategy(PersonalRandomNumberGenerator& rng, Person& p
 
     //lookup schemes for this specific location as well as the location type
     //lookup in std::vector instead of std::map should be much faster unless for large numbers of schemes
-    for (auto loc_key : {LocationId{location.get_index(), location.get_type()},
-                         LocationId{INVALID_LOCATION_INDEX, location.get_type()}}) {
+    for (auto key : {std::make_pair(location.get_type(), location.get_id()),
+                     std::make_pair(location.get_type(), LocationId::invalid_id())}) {
         auto iter_schemes =
-            std::find_if(m_location_to_schemes_map.begin(), m_location_to_schemes_map.end(), [loc_key](auto& p) {
-                return p.first == loc_key;
+            std::find_if(m_location_to_schemes_map.begin(), m_location_to_schemes_map.end(), [&](const auto& p) {
+                return p.type == key.first && p.id == key.second;
```

Comment:
What do you propose?
Since it is very local, the short name is fine by me.

Reply:
Short name is fine, it's just that `p` is usually used for persons. Perhaps `loc`?

Reply:
`loc` is not really accurate as well (and neither is/was `m_location_to_schemes_map`), since this is a triple of loc id, loc type and a vector of schemes. How about something generic like `x`? Otherwise I would consider reworking the internals here in a new issue (also see my comment below on omitting loc types).

Reply:
Right, perhaps `item` or `entry`? I'm fine with a separate issue.